### PR TITLE
util/mock: move mocks

### PR DESCRIPTION
**Pull request by coocood**
`mocks` has more dependency than `mock`, so they have to be in different packages to avoid import cycle, under the same package seems odds, so follow our convention, move `mocks` into `mock` package

### CODE REVIEW

File: util/mock/mocks/statement.go [hunk 1]
```diff
@@ -7,48 +7,48 @@ import (
 	"github.com/pingcap/tidb/util/format"
 )
 
-// MockStatement represents a mocked Statement.
-type MockStatement struct {
+// Statement represents a mocked Statement.
+type Statement struct {
```
Comment (qiuyesuifeng) [inline util/mock/mocks/statement.go]: Add license header.

**Comment by siddontang** [comment]
// Package mocks is just for test only.
package mocks

add a comment like mock.

**Comment by siddontang** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**